pkg/kritis/cryptolib/verifier_test.go

```diff
@@ -24,15 +24,123 @@ import (
+CQgLAgQWAgMBAh4BAheAAAoJEHe0vmmJg0Izes8L/1Xcxy0/6ziUu8HIuRT9ypyi
+AHpqmEuj9t0t+6mnOJPhbg4yySFUn5BuH2QWiS6SOWZzIKfjT9FtrooDxadM59Pd
+MmqR30PBfLh9Va4CDltDZqdmO+32ycz1M1ea1HYGXJgpk7gfjJGsW0+iEpbQJVWz
+pzSiqpYwpLTF+dCbWhNMUK4Yy2k0dBr/qU2pSiJvUNAE82/SqASB3sSHMQegO21b
+LaAotUWxpY362BqfPXwxzVE74+8FxLk54EGAfKDV0hlJRtWzk0++qR+VdthlxCLy
+hMQZtR/nsHBC2onaM/gsFQivcarSYmt+c9mgG55kr1kjnJrBaoIwkvBEGLHO0EHx
+h0KSmoi/kXuL8+sbzb90yn6O9HlC3ZvdffMbuHb+7SoIw62aWBWpkgovHdmIi9iQ
+wrtfH05UEkdZL3F3Gqn4OQxevoWw3IOxmDZX0vghpdpwvEMCF0btxXBuTqTyJz43
+DgFPbRPrOA0XZyVceVZ+gXavuHGIjvQShOUBlRlLELkBjQRe4R4XAQwAwt4ubiZB
+sfYsfRc0dneHEoDlirdDmhSfH1UXDvKJfpxE6dZOjyDR48wzm7nuwMH4bUaKs8Sh
+n0AMaVm1lhC/usppjs0FwmxnkxWID4qTZA0PQuvmmncQWCb74hDTocyB/YwXESxI
+hP9aK5Yny6Yz6mN+C+MUSH2GnC3Pn5LD7PkW8K8e7RFK7qBTtW+F/NsqGJ0cexQH
+2DR0d/mlrmgEIFRUJpKR3/c9miuG/U2Moa+qO3gQGSIBARyzCEfnjqmmjwGwOc0P
+ExrcgCict97UXRRh1SMpqUh9emYvCqd7wMlmrs7YQ/ke4ZLOg/3+iX1HYdS3wh/3
+Rnj6iMBRtgThYrQJpR3EOvjTM1TqN8nI1Dv+PjSWjcp9WDiT2jBErNoO7dBwMnb5
+UGgzZKx5yQy3BcXOTL3977tq/6HuoFeSszGkt49Kmcn2PgGEoHjEVkAFxn93UQyC
+LCD2rQ2iwlo8CxKaThbiGeGfSasda46HGnk4salKzsTQM30FPtMU5VY/ABEBAAGJ
+AbYEGAEKACAWIQREZiXrNgNtdUay1jt3tL5piYNCMwUCXuEeFwIbDAAKCRB3tL5p
+iYNCM99AC/9Wi+uYPR4QXcbXXc66YYJy/zx//rod8bjElAzbKp+aAafdK8Ygps/g
+OxDxXNZMQzrTSWUUnwLIgc+85BE4j91IeDMLrz1HWiYpO8uZHsWVIJ9OThyLf1pz
+NCrL3nhyZM94Gz630lO6qe89TawBKqq8jBbMQMLEWN8qq3dD7Hs+VK5LjL/1SM1V
+SKK1Eg9Up/Z4A8QBCMUoSWSlshAtlykQEhLECc3/omgcwHxnqIcIY/+b2b4ov6jT
+ZCNjOloppNjUhAhb6BTTL1sYqURw/cMmFgol9Fn1skVVwRv9Ss5mRt72M13k5SUr
+78/h+vRv8Fxz5X0Dx2sB7KMjFAn81rbeaHY/iC1jAQ/hyDiRNsGPnABfZF/WAPw8
+35I1pNHW36kciRPadnvxhMWLKEIp5DRzviETPBSX56RDUB6zHRYJzj1HhJg9rTMt
+j5UWIor77TvJrNtSBsAjCQW45RNqq7ktuQX8mLHdu1uoNVqNPCXZdHdrl7ykU82P
+w084byeM9nc=
+=zsax
+-----END PGP PUBLIC KEY BLOCK-----`
+
+const verifierPublicKeyID = "446625EB36036D7546B2D63B77B4BE6989834233"
+
+func TestNewPublicKey(t *testing.T) {
+	tcs := []struct {
+		name        string
+		keyType     KeyType
+		keyData     []byte
+		keyID       string
+		expectedErr bool
+	}{
+		{
+			name:        "valid PGP key ID",
+			keyType:     Pgp,
+			keyData:     []byte(verifierPublicKey),
+			keyID:       verifierPublicKeyID,
+			expectedErr: false,
+		},
+		{
+			name:        "invalid PGP key ID",
+			keyType:     Pgp,
+			keyData:     []byte(verifierPublicKey),
+			keyID:       "incorrect-id",
+			expectedErr: true,
+		},
+		{
+			name:        "valid PKIX key ID",
+			keyType:     Pkix,
+			keyID:       "valid-key-id",
+			expectedErr: false,
+		},
+		{
+			name:        "invalid PKIX key ID",
+			keyType:     Pkix,
+			keyID:       "^{invalid-key-id}",
+			expectedErr: true,
+		},
+		{
+			name:        "unknown key type",
+			keyType:     UnknownKeyType,
+			expectedErr: true,
+		},
+	}
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			_, err := NewPublicKey(tc.keyType, tc.keyData, tc.keyID)
+			if tc.expectedErr && err == nil {
+				t.Errorf("Expected error, but returned none")
```

Comment:
In go test error messages, put got before want, e.x. "got nil err, expected not-nil"

Reply:
Done.